src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/temperature/dimension/CpuUtilDimensionTemperatureRca.java

```diff
@@ -73,9 +73,15 @@ public DimensionalTemperatureFlowUnit operate() {
                 THRESHOLD_NORMALIZED_VAL_FOR_HEAT_ZONE_ASSIGNMENT);
         LOG.info("CPU Utilization temperature calculated: {}",
                 cpuUtilTemperatureFlowUnit.getNodeDimensionProfile());
-        ResourceContext context = (cpuUtilTemperatureFlowUnit.getNodeDimensionProfile().getMeanTemperature()
-                .isGreaterThan(THRESHOLD_NORMALIZED_VAL_FOR_HEAT_ZONE_ASSIGNMENT)) ? new ResourceContext(Resources.State.UNHEALTHY) :
-                new ResourceContext(Resources.State.HEALTHY);
+        ResourceContext context;
+        try {
+            context = (cpuUtilTemperatureFlowUnit.getNodeDimensionProfile().getMeanTemperature()
+                    .isGreaterThan(THRESHOLD_NORMALIZED_VAL_FOR_HEAT_ZONE_ASSIGNMENT)) ? new ResourceContext(Resources.State.UNHEALTHY) :
+                    new ResourceContext(Resources.State.HEALTHY);
```

Comment:
We should remove this as the health based on temperature is not inferred at this level. It happens at a higher level component - pyrometer.

Reply:
Done